elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala

```diff
@@ -936,6 +936,25 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
       restClient.documentExistsById(index, tpe, "doc0002").futureValue should be(false)
     }
 
+    "Support deleting more than 10000 docs" in {
+      val insertFutures = (1 to 10011).map(i => restClient.index(index, tpe, Document(s"doc$i", Map("text7" -> "here7"))))
+      val ir = Future.sequence(insertFutures)
+      Await.result(ir, 20.seconds)
+      refresh()
+
+      val delFut = restClient.deleteDocuments(index, tpe, new QueryRoot(MatchAll, sizeOpt = Some(9)))
```

Comment:
Can you also add a test that shows that the existing `deleteDocument` API does not work?

Reply:
sure, same test, just replace line 945:
val delFut = restClient.deleteDocuments(index, tpe, new QueryRoot(MatchAll, sizeOpt = Some(9)))
with:
val delFut = restClient.deleteDocument(index, tpe, new QueryRoot(MatchAll, sizeOpt = Some(10011)))

Should fail with an exception: "Result window is too large, from + size must be less than or equal to: [10000] but was [10011]."

But I'm not sure if we want to add such test to the codebase, since it's assuring feature failure, not success.